lib/apiservers/engine/backends/system.go

```diff
@@ -240,7 +241,8 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+	defer trace.End(trace.Begin(""))
+	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
```

Comment:
I can look into reducing the footprint.  We can also abstract the authen/authoriz code from imagec out and implement our own.  I used the docker code as the most expedient path to implementation for the upcoming deliverable.  Let me write some robot scripts to test the login/pull workflow, and then I can revisit this.